runtime/jcl/common/java_dyn_methodhandle.c

```diff
@@ -1045,3 +1045,14 @@ clearNonZAAPEligibleBit(JNIEnv *env, jclass nativeClass, const JNINativeMethod *
 	vmFuncs->internalExitVMToJNI(vmThread);
 }
 #endif /* J9VM_OPT_JAVA_OFFLOAD_SUPPORT */
+
+#if JAVA_SPEC_VERSION >= 15
+void JNICALL
+Java_java_lang_invoke_MethodHandleNatives_checkClassBytes(JNIEnv *env, jclass jlClass, jbyteArray classRep, jint length)
+{
+	J9VMThread *currentThread = (J9VMThread *)env;
+	J9InternalVMFunctions *vmFuncs = currentThread->javaVM->internalVMFunctions;
+	U_8* classBytes = (U_8*) TMP_J9JAVACONTIGUOUSARRAYOFBYTE_EA(currentThread, *(J9IndexableObject **)classRep, 0);
```

Comment:
In fact, it won't work at all (you're assuming you have VM access in a normal JNI native). Please use JNI critical instead. This will require changing checkClassBytes to not throw the exception (have a return code instead).

Reply:
Updated to use GetPrimitiveArrayCritical().